platform/admin/admin-configuration-configinstaller/src/main/java/org/codice/ddf/admin/configuration/ConfigurationInstaller.java

```diff
@@ -0,0 +1,201 @@
+    }
+
+    @Override
+    public void configurationEvent(ConfigurationEvent configurationEvent) {
+        switch (configurationEvent.getType()) {
+        case CM_UPDATED:
+            LOGGER.debug("Handling update");
+            Configuration configuration = getConfiguration(configurationEvent);
+            if (configuration == null) {
+                break;
+            }
+            handleUpdate(new FelixConfig(configuration));
+            break;
+        case CM_DELETED:
+            LOGGER.debug("Handling delete");
+            handleDelete(configurationEvent.getPid());
+            break;
+        default:
+            LOGGER.debug("Unknown configuration event type, taking no action");
+        }
+    }
+
+    /**
+     * Sync the config files in etc when an update occurs.
+     *
+     * @param config the {@link Configuration} wrapped in a {@link FelixConfig} that arrived as an event.
+     */
+    private void handleUpdate(FelixConfig config) {
+        String pid = config.getPid();
+        File felixFileNew = config.getFile();
+        File felixFileOriginal = pidFileMap.get(pid);
+
+        boolean filePropChanged = config.filePropChanged(felixFileOriginal);
+
+        if (felixFileOriginal == null && felixFileNew == null) {
+            LOGGER.debug("Was not tracked and will not track pid {}", pid);
+            return;
+        }
+
+        if (filePropChanged) {
+            if (felixFileOriginal != null) {
+                LOGGER.debug("Felix filename prop changed, deleting file for pid {}", pid);
+                performOperation(configActions.delete(felixFileOriginal.toPath()));
+                if (felixFileNew == null) {
+                    LOGGER.debug("No longer tracking file for pid {}", pid);
+                    pidFileMap.remove(pid);
+                    return;
+                }
+            }
+            LOGGER.debug("Tracking pid {}", pid);
+            pidFileMap.put(pid, felixFileNew);
+        }
+    }
+
+    /**
+     * Remove config files from etc when a delete occurs.
+     *
+     * @param pid the pid of the configuration that was deleted.
+     */
+    private void handleDelete(String pid) {
+        File felixFileOriginal = pidFileMap.remove(pid);
+        if (felixFileOriginal != null) {
+            LOGGER.debug("Deleting file because config was deleted for pid {}", pid);
+            performOperation(configActions.delete(felixFileOriginal.toPath()));
+        }
+    }
+
+    /**
+     * @param operation the operation to perform.
+     */
+    private void performOperation(Operation operation) {
+        Configurator configurator = configuratorFactory.getConfigurator();
+        configurator.add(operation);
+        OperationReport report = configurator.commit("Synchronizing the etc directory");
+        if (report.containsFailedResults()) {
+            LOGGER.warn("One or more file operations failed, see debug logs for more info");
+            report.getFailedResults()
```

Comment:
I think it is fine. I would follow Richard's recommendation of wrapping the stream. You can refer to our logging guidelines here: https://codice.atlassian.net/wiki/display/DDF/Log+Levels but we do this pattern in quite a few other places of the code to avoid polluting the log files with noise: https://github.com/codice/ddf/blob/1784406cbf9872c2d03753f5571eaf8a1b053be1/catalog/ui/search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/controller/search/QueryRunnable.java#L279